### PR TITLE
chore: setup jest for use with Typescript tests

### PR DESCRIPTION
**Pull request by karrui**
This PR adds the ability to run Typescript tests with Jest. All tests with file names that match 

```regex
**/?(*.)+(spec|test).[t]s?(x)
``` 

will be run with Jest.

This PR also migrates a few tests to Typescript to check if `ts-jest` is working.

**Features**:
- A small lint improvement has been made to the autosorting and autogrouping of imports.
- Typescript tests are now ran with Jest.
- Update `tsconfig.json` to now allow for absolute imports: `../../../../../<root>/some/path` -> `src/some/path`
- Add `jest-mongodb-config`. With this, we can eventually stop using `npm run download-binary` and delete `tests/end-to-end/helpers/get-mongo-binary.js` since the config automatically retrieves that information for us.

**New scripts**:
- Add `test-backend-jest` script to run jest tests
- `test-backend` script now runs both jasmine and jest tests. Jasmine tests will be run first, followed by jest tests. Eventually, all tests will be written in Jest and Jasmine tests can be deleted.

### CODE REVIEW

**Comment by karrui** [comment]
Drafting this, I want to explore using Jest's automatic mongodb setup.

**Comment by karrui** [comment]
Ready for review. I think tests can be written in Typescript moving forward :)

File: jest-mongodb-config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  mongodbMemoryServerOptions: {
+    binary: {
+      version: process.env.MONGO_BINARY_VERSION || '3.6.12',
+      skipMD5: true,
```
Comment (liangyuanruo) [inline jest-mongodb-config.js:5]: i think we should verify the authenticity of the binary, it shouldn't take that much time

**Comment by karrui** [comment]
Yes. But also need to change all Jasmine usages to jest, and update the imports also

Best regards,
Kar Rui
________________________________

**Pull request merged**